src/main/scala/inox/solvers/smtlib/Z3Target.scala

```diff
@@ -57,9 +57,8 @@ trait Z3Target extends SMTLIBTarget with SMTLIBDebugger {
   protected val interpreter = {
     val opts = interpreterOpts
     reporter.debug("Invoking solver "+targetName+" with "+opts.mkString(" "))
-    new Z3Interpreter(targetName, opts.toArray) {
-      override lazy val parser: Z3Parser = new Z3Parser(new Lexer(out))
-    }
+    class InterpreterImpl extends Z3Interpreter(targetName, opts.toArray, parserCtor = out => new Z3Parser(new Lexer(out)))
+    new InterpreterImpl
```

Comment:
Can't you just do `new Z3Interpreter(...)` here?

Reply:
Whoops, my bad, thanks for pointing that out!